xbmc/cores/AudioEngine/Sinks/AESinkDARWINOSX.cpp

```diff
@@ -93,13 +93,56 @@ static CADeviceList GetDevices()
   return list;
 }
 
+//---------- DEVICE CHANGED EVENT HANDLING
+// when device list change callbacks are received
+// inform the ae engine this amount of time after
+// the last callback was received (bundle multiple callbacks)
+// into one final call
+#define DEVICE_LIST_CHANGED_TIMER_MS 1500
+
+CDeviceChangeListTimeout::CDeviceChangeListTimeout()
+{
+  m_pDeviceListChangedTimer = new CTimer(this);
```

Comment:
Basically your code suppresses callbacks for 1,5 secs after the first callback. My code delays the final callback until no events occured for 1,5seconds ...